### PR TITLE
Final round of `db.DefaultContext` refactor

### PR DESCRIPTION
**Pull request by JakobDev**
Last part of #27065

### CODE REVIEW

File: models/asymkey/gpg_key_verify.go [hunk 1]
```diff
@@ -29,8 +30,8 @@ import (
 // This file provides functions relating verifying gpg keys
 
 // VerifyGPGKey marks a GPG key as verified
-func VerifyGPGKey(ownerID int64, keyID, token, signature string) (string, error) {
-	ctx, committer, err := db.TxContext(db.DefaultContext)
+func VerifyGPGKey(ctx context.Context, ownerID int64, keyID, token, signature string) (string, error) {
+	ctx, committer, err := db.TxContext(ctx)
```
Comment (KN4CK3R) [inline models/asymkey/gpg_key_verify.go:34]:
```suggestion
	dbCtx, committer, err := db.TxContext(ctx)
```

File: models/asymkey/ssh_key_verify.go [hunk 2]
```diff
@@ -13,8 +14,8 @@ import (
 )
 
 // VerifySSHKey marks a SSH key as verified
-func VerifySSHKey(ownerID int64, fingerprint, token, signature string) (string, error) {
-	ctx, committer, err := db.TxContext(db.DefaultContext)
+func VerifySSHKey(ctx context.Context, ownerID int64, fingerprint, token, signature string) (string, error) {
+	ctx, committer, err := db.TxContext(ctx)
```
Comment (KN4CK3R) [inline models/asymkey/ssh_key_verify.go:18]:
```suggestion
	dbCtx, committer, err := db.TxContext(ctx)
```

File: models/auth/oauth2.go [hunk 3]
```diff
@@ -284,8 +284,8 @@ type UpdateOAuth2ApplicationOptions struct {
 }
 
 // UpdateOAuth2Application updates an oauth2 application
-func UpdateOAuth2Application(opts UpdateOAuth2ApplicationOptions) (*OAuth2Application, error) {
-	ctx, committer, err := db.TxContext(db.DefaultContext)
+func UpdateOAuth2Application(ctx context.Context, opts UpdateOAuth2ApplicationOptions) (*OAuth2Application, error) {
+	ctx, committer, err := db.TxContext(ctx)
```
Comment (KN4CK3R) [inline models/auth/oauth2.go:288]:
```suggestion
	dbCtx, committer, err := db.TxContext(ctx)
```

File: models/auth/oauth2.go [hunk 4]
```diff
@@ -352,8 +352,8 @@ func deleteOAuth2Application(ctx context.Context, id, userid int64) error {
 }
 
 // DeleteOAuth2Application deletes the application with the given id and the grants and auth codes related to it. It checks if the userid was the creator of the app.
-func DeleteOAuth2Application(id, userid int64) error {
-	ctx, committer, err := db.TxContext(db.DefaultContext)
+func DeleteOAuth2Application(ctx context.Context, id, userid int64) error {
+	ctx, committer, err := db.TxContext(ctx)
```
Comment (KN4CK3R) [inline models/auth/oauth2.go:356]:
```suggestion
	dbCtx, committer, err := db.TxContext(ctx)
```

File: services/pull/lfs.go [hunk 5]
```diff
@@ -40,7 +40,7 @@ func LFSPush(ctx context.Context, tmpBasePath, mergeHeadSHA, mergeBaseSHA string
 	// 6. Take the output of cat-file --batch and check if each file in turn
 	// to see if they're pointers to files in the LFS store associated with
 	// the head repo and add them to the base repo if so
-	go createLFSMetaObjectsFromCatFileBatch(catFileBatchReader, &wg, pr)
+	go createLFSMetaObjectsFromCatFileBatch(db.DefaultContext, catFileBatchReader, &wg, pr)
```
Comment (delvh) [inline services/pull/lfs.go:43]: Why `default` if we have a `ctx` param?
```suggestion
	go createLFSMetaObjectsFromCatFileBatch(ctx, catFileBatchReader, &wg, pr)
```

**Review by delvh — APPROVED** [review]

File: models/asymkey/gpg_key_verify.go (hunk 1 above)
Comment (JakobDev) [reply models/asymkey/gpg_key_verify.go:34]: I think the rue is, the don't rename `ctx `to `dbCtx` if `ctx` is not used after the transaction is closed

File: services/pull/lfs.go (hunk 5 above)
Comment (JakobDev) [reply services/pull/lfs.go:43]: This is a goroutine, which could cause problems

**Review by KN4CK3R — APPROVED** [review]

**Pull request merged**